aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp

```diff
@@ -218,6 +219,92 @@ using namespace at::sparse_csr;
 // certain utiliy functions are usable from sparse COO.
 using namespace at::sparse;
 
+Tensor& mul_out_sparse_csr(const Tensor& t_, const Tensor& src_, Tensor& r) {
+  // // TODO: Use a specialized CSR kernel for performance if needed
+  if (t_.is_sparse_csr() && src_.layout() == kStrided) {
+    return mul_out_sparse_csr(t_, src_.sparse_mask(t_), r);
+  }
+  if (t_.layout() == kStrided && src_.is_sparse_csr()) {
+    return mul_out_sparse_csr(t_.sparse_mask(src_), src_, r);
+  }
+  TORCH_CHECK(r.is_sparse_csr(), "Expected result Tensor to be of format CSR");
```

Comment:
Just a note for a possible follow-up: generalizing this function to other sparse compressed layouts looks like a straightforward task. It would remove one of the blockers in test_gradcheck_sparse_csc_input of test/test_autograd.py.